pramen-py/src/pramen_py/metastore/writer.py

```diff
@@ -41,25 +43,50 @@ def write(
         logger.info(f"Writing table {table_name} started")
 
         target_table = get_metastore_table(table_name, self.config)
-        target_path = os.path.join(
-            target_table.path,
-            f"{target_table.info_date_settings.column}={self.info_date}",
-        )
-        df = df.drop(target_table.info_date_settings.column)
-        df_writer = (
-            # TODO #24 calculate and make repartition instead
-            df.write.option(
-                "maxRecordsPerFile",
-                target_table.records_per_partition,
-            )
-        )
-        if target_table.format is TableFormat.parquet:
-            df_writer.parquet(target_path, mode="overwrite")
-        elif target_table.format is TableFormat.delta:
-            raise NotImplementedError
+        if target_table.format == TableFormat.parquet:
+            save_path = self._write_parquet_format_table(df, target_table)
+        elif target_table.format == TableFormat.delta:
+            save_path = self._write_delta_format_table(df, target_table)
         else:
             raise NotImplementedError
         logger.info(
             f"Successfully written {df.count()} items to {table_name} at "
```

Comment:
`df.count()` is a very expensive operation. I see it is invoked more than once when writing data.
Please, create a variable, and reuse the result in several places in the code.